### ISSUE TITLE
Adding a whitelist preference?

### ISSUE THREAD

**Issue by petertirrell** [ram-on/SkyTube#193]
I'm trying to solve a problem where I'd like a youtube app that only allows specific channels, or specific video links, or specific playlists.  I came across this app, and am wondering about adding a whitelist URL list in the "Other Preferences" and then have it only display those videos if there is anything in the whitelist section.  

I'm going to try and pull down the code this weekend and maybe poke at it some, but are there any thoughts on such a thing in the first place?

Are there any developer guidelines or instructions?  Is it just a matter of opening the project in Android Studio and it should all just build?

**Comment by atomjack** [comment]
You should fork the Repository and then pull your fork down, and then do any work you want on a new branch. That way if you want to add stuff to the app you can push your branch up and make a Pull Request.

One thing you will have to do in order to make the app build is this, once you pull down your fork:

```
cd app/src/main/java/org/schabi/newpipe/
git clone https://github.com/ram-on/extractor.git
```

The app uses this Extractor code from the NewPipe project, and it won't build without it.

**Comment by ram-on** [comment]
@petertirrell I was more planning to develop a black list (rather than a white list) in the sense that certain channels will be blocked.

The problem with channel whitelist is that it will make the app unusable as the absolute majority of the videos will be blocked by default -- even worse with video whitelisting.

What I suggest is that you implement channel blacklisting.  I would really appreciate it as it is in the pipeline.

As of developer guidelines:

* Each class and method shall be properly documented.
* Code shall be maintainable -- no code copy and pasting.

Also, pls follow @atomjack instructions.

**Comment by petertirrell** [comment]
Thanks for the feedback!  My whitelist idea actually stems from wanting to block the majority of videos out there...I'm trying to find a better solution for Youtube parental controls than what I've found so far.  Youtube's preferences don't really offer much there, and even the YouTube Kids app still offers more than I'd prefer, and still doesn't really have much in the way of filtering options.  

So I had the thought that if I could 'OK' specific channels, playlists, or specific video URLs, that could solve what I'm looking for.  I can see the advantages of a blacklist for sure, though, too.  

Either way, I'll make a fork and play around with the codebase a bit just to get a feel for it anyway.  Thanks again,

**Comment by okan35** [comment]
@ram-on Hey Ramon, I was just going to open an issue about blacklisting the channels as I have been using your app for a long time I would like to work on this feature as much as I can. Can I start it immediately ?

**Comment by ram-on** [comment]
@okan35 I would appreciate if channel blacklisting is implemented.

As for the Turkish translation, these can be performed on the weblate website:  https://hosted.weblate.org/engage/skytube/

**Comment by ram-on** [comment]
@petertirrell Channel blacklisting implemented as per PR #248

**Comment by ram-on** [comment]
@petertirrell Channel whitelisting implemented as per 5e6b8d15715ba11498ce16f5dae47f4de5ab982c.
